subsys/bluetooth/mesh/time_util.h

```diff
@@ -20,8 +20,6 @@ extern "C" {
 
 #define DAYS_YEAR 365ULL
 #define DAYS_LEAP_YEAR 366ULL
-#define SEC_PER_HOUR (60ULL * SEC_PER_MIN)
```

Comment:
I guess Zephyr's definitions have type `ul` 4 bytes and using them in time calculation might overflow the final result, for instance here: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/time_util.c#L44
`struct tm` considers year since 1900. Then seconds for dates in 2025 will be 125 * 366 * 60 * 60 * 24 = 3952800000 = 0xEB9AF100
Overflow will happen in about 2035-2036.

Reply:
Good catch, thanks! I was aware of this change in type and I looked for potential overflows, but apparently I haven't done it thoroughly enough for `SEC_PER_DAY` and missed that one problematic place.